aten/src/ATen/function_wrapper.py

```diff
@@ -609,6 +619,17 @@ def to_return_type(arg, option):
     }
 
 
+def emit_version_increment(option):
+    version_increment_stmts = []
+    for formal in option['formals_list']:
+        arg_type = formal['type']
+        if NATIVE_DYNAMIC_TYPE.get(arg_type, arg_type) == 'Tensor' and 'const' not in arg_type:
```

Comment:
is NATIVE_DYNAMIC_TYPE safe with BoolTensor, etc.?

Reply:
Yes it is safe with BoolTensor / IndexTensor / IntegerTensor. Examples:
```python
# BoolTensor:
{
 'api_name': '_th_lt_out',
 'formals_list': [
    {
        'dynamic_type': 'BoolTensor',
        'name': 'result',
        'output': True,
        'type': 'Tensor &'  # formal['type'] of a BoolTensor is `Tensor &`, which is what we expected
    },
    ...]
}

# IndexTensor
{
 'api_name': '_th_nonzero_out',
 'formals_list': [
    {
        'dynamic_type': 'IndexTensor',
        'name': 'result',
        'output': True,
        'type': 'Tensor &'  # formal['type'] of an IndexTensor is `Tensor &`, which is what we expected
   },
   …],
}

# IntegerTensor
{
 'api_name': '_th_pstrf_out',
 'formals_list': [...,
    {
        'dynamic_type': 'IntegerTensor',
        'name': 'res2',
        'output': True,
        'type': 'Tensor &'  # formal['type'] of an IntegerTensor is `Tensor &`, which is what we expected
   },
   …],
}

```